src/window.rs

```diff
@@ -363,14 +363,69 @@ fn stringify_message_content(message: &TelegramMessage, chat: &Chat) -> String {
                 gettext("Channel photo removed")
             }
             _ => {
-                if message.is_outgoing {
-                    gettext("You removed the group photo")
-                } else {
-                    let sender_name = sender_name(&message.sender_id, chat);
-                    gettext!("{} removed the group photo", sender_name)
-                }
+                let sender_name = sender_name(&message.sender_id, chat);
+                gettext!("{} removed the group photo", sender_name)
+            }
+        },
+        MessageContent::MessageChatChangePhoto(_) => match chat.type_() {
+            ChatType::Supergroup(data) if data.is_channel() => gettext("Channel photo changed"),
+            _ => {
+                gettext!(
+                    "{} changed group photo",
+                    sender_name(&message.sender_id, chat),
+                )
             }
         },
+        MessageContent::MessagePinMessage(data) => {
+            let message_content = match chat.history().message_by_id(data.message_id) {
+                Some(data) => match data.content().0 {
+                    MessageContent::MessageText(data) => {
+                        let msg = data.text.text;
+                        if msg.chars().count() > 17 {
+                            gettext!("«{}»", msg.chars().take(16).collect::<String>())
```

Comment:
Shouldn't the 3 dots `…` not be in this if branch instead of being in the else branch?